generators/server/index.js

```diff
@@ -1021,6 +1021,37 @@ module.exports = JhipsterServerGenerator.extend({
                     this.copy(SERVER_MAIN_RES_DIR + 'config/cql/changelog/_insert_default_users.cql', SERVER_MAIN_RES_DIR + 'config/cql/changelog/00000000000000_insert_default_users.cql');
                 }
             }
+
+            if (this.applicationType === 'uaa') {
+                const keyStoreFile = SERVER_MAIN_RES_DIR + 'keystore.jks';
+                if (this.fs.exists(keyStoreFile)) {
+                    this.log(chalk.cyan('\nKeyStore \'' + keyStoreFile + '\' already exists. Leaving unchanged.\n'));
+                } else {
+                    this.spawnCommand('keytool', [
+                        '-genkey',
+                        '-noprompt',
+                        '-keyalg', 'RSA',
+                        '-alias', 'selfsigned',
+                        '-keystore', keyStoreFile,
+                        '-storepass', 'password',
+                        '-keypass', 'password',
+                        '-keysize', '2048',
+                        '-dname', 'CN=Java Hipster, OU=Development, O=' + this.packageName + ', L=, ST=, C='
+                    ])
+                    .on('close', (code) => {
+                        if (code === 0) {
+                            this.log(chalk.green('\nKeyStore \'' + keyStoreFile + '\' generated successfully.\n'));
+                        } else {
+                            this.log(chalk.red(`\nFailed to create a KeyStore with \'keytool\'`));
+                            if (code === -2) {
+                                this.log(chalk.red('Is \'keytool\' available on the path along with your JDK installation? (i.e. JAVA_HOME/bin)'));
+                            }
+                            shelljs.exit(code);
+                        }
+                    })
+                    .on('error', (err) => {}); // must handle error if spawn failed but 'close' is also triggered.
```

Comment:
The yeoman spawnCommand will trigger 'error' if keytool is not installed. It also triggers 'close' as well though. If I don't have anything for 'error' then it throws an 'unhandled' exception. But 'error' is not enough because exit code 1 (which happens if keytool runs but doesn't generate anything) does not trigger 'error'...only 'close'. :P maybe shelljs would be cleaner?